index.js

```diff
@@ -79,59 +120,99 @@ function constructRequest(har, userAgent = false) {
+            //
+            // https://github.com/form-data/form-data/issues/124
+            throw new Error(
+              "We've detected you're using a non-spec compliant FormData library. We recommend polyfilling FormData with https://npm.im/formdata-node"
+            );
+          }
 
           request.postData.params.forEach(param => {
-            if ('fileName' in param && !('value' in param)) {
-              throw new Error(
-                "The supplied HAR has a postData parameter with `fileName`, but no `value` content. Since this library doesn't have access to the filesystem, it can't fetch that file."
-              );
-            }
-
-            // If the incoming parameter is a file, and that files value is a data URL, we should decode that and set
-            // the contents of the value in the HAR to the actual contents of the file.
             if ('fileName' in param) {
-              const parsed = parseDataUrl(param.value);
-              if (parsed) {
-                // eslint-disable-next-line no-param-reassign
-                param.value = parsed.toBuffer().toString();
-              }
-            }
+              if (opts.files && param.fileName in opts.files) {
+                const fileContents = opts.files[param.fileName];
+
+                // If the file we've got available to us is a Buffer then we need to convert it so that the FormData
+                // API can use it.
+                if (Buffer.isBuffer(fileContents)) {
+                  form.set(
+                    param.name,
+                    new File([fileContents], param.fileName, {
+                      type: param.contentType || null,
+                    }),
+                    param.fileName
+                  );
+
+                  return;
+                } else if (fileContents instanceof File) {
+                  // The `Blob` polyfill on Node comes back as being an instanceof `File`. Because passing a Blob into
+                  // a File will end up with a corrupted file we want to prevent this.
+                  //
+                  // This object identity crisis does not happen in the browser.
+                  if (fileContents.constructor.name === 'File') {
+                    form.set(param.name, fileContents, param.fileName);
+                    return;
+                  }
+                }
+
+                throw new TypeError(
+                  'An unknown object has been supplied into the `files` config for use. We only support instances of the File API and Node Buffer objects.'
+                );
+              } else if ('value' in param) {
+                let paramBlob;
+                const parsed = parseDataUrl(param.value);
+                if (parsed) {
+                  // If we were able to parse out this data URL we don't need to transform its data into a buffer for
+                  // `Blob` because that supports data URLs already.
```

Comment:
We were previously doing `param.value = parsed.toBuffer().toString();` which was corrupting image uploads.